src/api/MainSolver.h

```diff
@@ -35,19 +34,21 @@ class sstat {
             value = -1;
         else if (l == l_Undef)
             value = 0;
-        else assert(false);
+        else
+            assert(false);
     }
-    char getValue()            const { return value; }
+    char getValue() const { return value; }
     friend sstat toSstat(int v);
 };
 
-inline sstat toSstat(int v) {return sstat(v); }
+inline sstat toSstat(int v) {
+    return sstat(v);
+}
 
-const sstat s_True  = toSstat( 1);
+const sstat s_True = toSstat(1);
 const sstat s_False = toSstat(-1);
-const sstat s_Undef = toSstat( 0);
-const sstat s_Error = toSstat( 2);
-
+const sstat s_Undef = toSstat(0);
+const sstat s_Error = toSstat(2);
 
 class MainSolver {
 protected: /** Helper classes to deal with assertion stack, preprocessing and substitutions **/
```

Comment:
I would recommend to follow the order public, protected, private in general.
In most cases, when we look into header files, we are interested in the public API.
But this way, we have to scroll down and search for public parts.

In most cases, it is not a problem when a member that has not been defined yet is accessed.
It is not the case of member types and classes, but still public API should not depend on protected/private types.